src/overlays/actors/ovl_En_Yb/z_en_yb.c

```diff
@@ -170,31 +170,33 @@ void EnYb_ActorShadowFunc(Actor* thisx, Lights* mapper, PlayState* play) {
 }
 
 void EnYb_ChangeAnim(PlayState* play, EnYb* this, s16 animIndex, u8 animMode, f32 morphFrames) {
-    if ((animIndex >= 0) && (animIndex < 3)) {
-        if ((animIndex != this->animIndex) || (animMode != ANIMMODE_LOOP)) {
-            if (animIndex > 0) {
-                if (animMode == ANIMMODE_LOOP) {
-                    PlayerAnimation_Change(play, &this->skelAnime, gPlayerAnimations[animIndex - 1], 1.0f, 0.0f,
-                                           Animation_GetLastFrame(gPlayerAnimations[animIndex - 1]), ANIMMODE_LOOP,
-                                           morphFrames);
-                } else {
-                    // unused case, (only called once with animMode = ANIMMODE_LOOP)
-                    PlayerAnimation_Change(play, &this->skelAnime, gPlayerAnimations[animIndex - 1], 1.0f, 0.0f,
-                                           Animation_GetLastFrame(gPlayerAnimations[animIndex - 1]), ANIMMODE_LOOP,
-                                           morphFrames);
-                }
-            } else {
-                // unused case, (only called once with animIndex = 2)
-                AnimationHeader* animationPtr = gYbUnusedAnimations[animIndex];
+    if ((animIndex < 0) || (animIndex > 2)) {
+        return;
+    }
 
-                if (1) {}
+    if ((animIndex == this->animIndex) && (animMode == ANIMMODE_LOOP)) {
+        return;
+    }
 
-                Animation_Change(&this->skelAnime, gYbUnusedAnimations[animIndex], 1.0f, 0.0f,
-                                 Animation_GetLastFrame(animationPtr), animMode, morphFrames);
-            }
-            this->animIndex = animIndex;
+    if (animIndex > 0) {
+        if (animMode == ANIMMODE_LOOP) {
+            PlayerAnimation_Change(play, &this->skelAnime, gPlayerAnimations[animIndex - 1], 1.0f, 0.0f,
+                                   Animation_GetLastFrame(gPlayerAnimations[animIndex - 1]), ANIMMODE_LOOP,
+                                   morphFrames);
+        } else {
+            // unused case, (only called once with animMode = ANIMMODE_LOOP)
+            PlayerAnimation_Change(play, &this->skelAnime, gPlayerAnimations[animIndex - 1], 1.0f, 0.0f,
+                                   Animation_GetLastFrame(gPlayerAnimations[animIndex - 1]), ANIMMODE_LOOP,
+                                   morphFrames);
         }
+    } else {
+        // unused case, (only called once with animIndex = 2)
+        AnimationHeader* animationPtr = gYbUnusedAnimations[animIndex];
+
+        Animation_Change(&this->skelAnime, gYbUnusedAnimations[animIndex], 1.0f, 0.0f,
+                         Animation_GetLastFrame(animationPtr), animMode, morphFrames);
```

Comment:
```suggestion
        Animation_Change(&this->skelAnime, gYbUnusedAnimations[animIndex], 1.0f, 0.0f,
                         Animation_GetLastFrame(gYbUnusedAnimations[animIndex]), animMode, morphFrames);
```
This seems to work locally (though need to add a pad somewhere).